app/models/spree/order_contents_decorator.rb

```diff
@@ -1,5 +1,26 @@
 module Spree
   OrderContents.class_eval do
+
+    def add_to_line_item(variant, quantity, options = {})
+      line_item = grab_line_item_by_variant(variant, false, options)
+
+      if line_item && part_variants_match(line_item, variant, quantity, options)
+        line_item.quantity += quantity.to_i
+        line_item.currency = currency unless currency.nil?
```

Comment:
I see. In that case I think it's okay to leave it as-is. I don't want us to take a risk in case the code changes in the future.